snap/info.go

```diff
@@ -89,6 +89,35 @@ func MinimalPlaceInfo(name string, revision Revision) PlaceInfo {
 	return &Info{SideInfo: SideInfo{RealName: storeName, Revision: revision}, InstanceKey: instanceKey}
 }
 
+// ParsePlaceInfoFromSnapFileName returns a PlaceInfo with just the location
+// information for a snap of file name, failing if the snap file name is invalid
+// This explicitly does not support filenames with instance names in them
+func ParsePlaceInfoFromSnapFileName(sn string) (PlaceInfo, error) {
+	if sn == "" {
+		return nil, fmt.Errorf("empty snap file name")
+	}
+	if strings.Count(sn, "_") > 1 {
+		// too many "_", probabl has an instance key in the filename like in
+		// snap-name_key_23.snap
+		return nil, fmt.Errorf("too many '_' in snap file name")
+	}
+	idx := strings.IndexByte(sn, '_')
+	switch {
+	case idx < 0:
+		return nil, fmt.Errorf("snap file name %q has invalid format (missing '_')", sn)
+	case idx == 0:
+		return nil, fmt.Errorf("snap file name %q has invalid format (no snap name before '_')", sn)
+	}
+	// ensure that _ is not the last element
+	name := sn[:idx]
+	revnoNSuffix := sn[idx+1:]
+	rev, err := ParseRevision(strings.TrimSuffix(revnoNSuffix, ".snap"))
+	if err != nil {
+		return nil, err
```

Comment:
maybe prefix with  "cannot parse revision in snap file name %q: ... " ?

Reply:
sure, that makes sense